core/src/apps/ethereum/layout.py

```diff
@@ -25,69 +25,95 @@
 async def require_confirm_tx(
     to_bytes: bytes,
     value: int,
-    gas_price: int,
-    gas_limit: int,
+    maximum_fee: str,
+    fee_info_items: Iterable[tuple[str, str]],
     network: EthereumNetworkInfo,
     token: EthereumTokenInfo | None,
     chunkify: bool,
 ) -> None:
+    from trezor.ui.layouts import confirm_ethereum_tx
+
     if to_bytes:
         to_str = address_from_bytes(to_bytes, network)
     else:
         to_str = TR.ethereum__new_contract
         chunkify = False
 
     total_amount = format_ethereum_amount(value, token, network)
-    maximum_fee = format_ethereum_amount(gas_price * gas_limit, None, network)
-    gas_limit_str = TR.ethereum__units_template.format(gas_limit)
-    gas_price_str = format_ethereum_amount(
-        gas_price, None, network, force_unit_gwei=True
-    )
-
-    items = (
-        (TR.ethereum__gas_limit, gas_limit_str),
-        (TR.ethereum__gas_price, gas_price_str),
-    )
 
     await confirm_ethereum_tx(
-        to_str, total_amount, maximum_fee, items, chunkify=chunkify
+        to_str, total_amount, maximum_fee, fee_info_items, chunkify=chunkify
     )
 
 
-async def require_confirm_tx_eip1559(
-    to_bytes: bytes,
+async def require_confirm_stake(
+    addr_bytes: bytes,
     value: int,
-    max_gas_fee: int,
-    max_priority_fee: int,
-    gas_limit: int,
+    maximum_fee: str,
+    fee_info_items: Iterable[tuple[str, str]],
     network: EthereumNetworkInfo,
-    token: EthereumTokenInfo | None,
     chunkify: bool,
 ) -> None:
-    if to_bytes:
-        to_str = address_from_bytes(to_bytes, network)
-    else:
-        to_str = TR.ethereum__new_contract
-        chunkify = False
 
-    total_amount = format_ethereum_amount(value, token, network)
-    maximum_fee = format_ethereum_amount(max_gas_fee * gas_limit, None, network)
-    gas_limit_str = TR.ethereum__units_template.format(gas_limit)
-    max_gas_fee_str = format_ethereum_amount(
-        max_gas_fee, None, network, force_unit_gwei=True
-    )
-    max_priority_fee_str = format_ethereum_amount(
-        max_priority_fee, None, network, force_unit_gwei=True
+    addr_str = address_from_bytes(addr_bytes, network)
+    total_amount = format_ethereum_amount(value, None, network)
+    await confirm_ethereum_staking_tx(
```

Comment:
I think we could remove majority (maybe all) of the kwargs to save some storage bytes.

It is true that there is a lot of args, so kwargs have documentation function, but we generally do not use them where not necessary.

(Description can be achieved also with args, e.g. like this:
```python
await confirm_ethereum_staking_tx(
    TR.ethereum__staking_unstake,  # title
    TR.ethereum__staking_unstake_intro,  # intro_question
    TR.ethereum__staking_unstake,  # verb
    ...
)
```
)

Reply:
Ok, thanks, I kept the names of all the arguments in the comments for consistency.
https://github.com/trezor/trezor-firmware/pull/3534/commits/b3eb30b259e2034a10e613d2f20c7761538a91f7